cardano-api/src/Cardano/Api/Value.hs

```diff
@@ -158,6 +169,44 @@ lovelaceToValue :: Lovelace -> Value
 lovelaceToValue = Value . Map.singleton AdaAssetId . lovelaceToQuantity
 
 
+toMaryValue :: forall ledgerera.
+               Ledger.Crypto ledgerera ~ StandardCrypto
+            => Value -> Mary.Value ledgerera
+toMaryValue v =
+    Mary.Value lovelace other
+  where
+    Quantity lovelace = selectAsset v AdaAssetId
+      --TODO: write QC tests to show it's ok to use Map.fromAscListWith here
+    other = Map.fromListWith Map.union
+              [ (toMaryPolicyID pid, Map.singleton (toMaryAssetName name) q)
+              | (AssetId pid name, Quantity q) <- valueToList v ]
+
+    toMaryPolicyID :: PolicyId -> Mary.PolicyID ledgerera
+    toMaryPolicyID (PolicyId sh) = Mary.PolicyID (toShelleyScriptHash sh)
+
+    toMaryAssetName :: AssetName -> Mary.AssetName
+    toMaryAssetName (AssetName n) = Mary.AssetName n
+
+
+fromMaryValue :: forall ledgerera.
+                 Ledger.Crypto ledgerera ~ StandardCrypto
+              => Mary.Value ledgerera -> Value
+fromMaryValue (Mary.Value lovelace other) =
+    Value $
+      --TODO: write QC tests to show it's ok to use Map.fromAscList here
```

Comment:
And this would be okay because we're deriving an `Ord` instance for `AssetId` in which the earliest constructor declared is `AdaAssetId`, right?

So an appropriate test would basically just verify that `AdaAssetId < AssetId x y`.

Reply:
Both order of ada vs other assets, and order amongst non-ada assets. This should be the case because converting to/from the ledger lib PolicyId and AssetName should not affect the order.

> So an appropriate test would basically just verify that AdaAssetId < AssetId x y

No, it'd be the whole order of converting the value to a list, including all assets (including cases of assets with a common policy id).